odiglet/pkg/kube/runtime_details/pods_controller.go

```diff
@@ -18,28 +19,74 @@ import (
 type podPredicate struct{}
 
 func (p *podPredicate) Create(e event.CreateEvent) bool {
-	// this function is called when a new entity is created in the controller-runtime cache.
-	// 1. When odiglet restarts, it will receive create events for all running pods.
-	// 2. When a new pod is created in k8s, should not have status phase as running, but it does not harm to check.
 	pod, ok := e.Object.(*corev1.Pod)
 	if !ok {
 		return false
 	}
-	return pod.Status.Phase == corev1.PodRunning
+
+	// Check if pod is in Running phase. This is the first requirement
+	if pod.Status.Phase != corev1.PodRunning {
+		return false
+	}
+
+	// If pod has no containers, return false as we can't determine readiness
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return false
+	}
+
+	// Iterate over all containers in the pod
+	// Return false if any container is:
+	// 1. Not Ready
+	// 2. Started is nil or false
+	for _, containerStatus := range pod.Status.ContainerStatuses {
```

Comment:
we do not check the previous event here, thus after this state is achieved, every new update will trigger language detection.
We should compare old and new value and only trigger when not all containers are ready but now they are